### ISSUE TITLE
Dates are filtered out

### ISSUE THREAD

**Issue by mkornatz** [cyrilschumacher/json-property-filter#1]
Hey there, I ran into an issue when implementing this library in an app I'm working on. Date objects are being filtered out unexpectedly, even when explicitly being included in the filter. I realize that this is a "JSON" filter, and date objects technically aren't allowed in JSON. So, this may be intended behavior. If that's the case, maybe it's worth adding a bit to the README about that.

Here's an example to show how to reproduce my issue:
```
const JsonPropertyFilter = require('json-property-filter');
const filter = new JsonPropertyFilter.JsonPropertyFilter([
  '**',
  '-id',
  'myDate'
]);

const myObject = {
  id: '123',
  myDate: new Date('2017-08-22T19:35:05.000Z'),
  subObj: {
    mySubDate: new Date('2017-08-22T19:35:05.000Z')
  }
};

const filtered = filter.apply(myObject);

console.log(myObject);
/**
  Outputs:
  {
    id: '123',
    myDate: 2017-08-22T19:35:05.000Z,
    subObj: {
      mySubDate: 2017-08-22T19:35:05.000Z
    }
  }
 */

console.log(filtered);
/**
 * Outputs: {}
 */
```

To work around the issue, here's what I'm doing:
```
const parsed = JSON.parse(JSON.stringify(myObject));
const filtered = filter.apply(parsed);
/**
  Outputs:
  {
    myDate: 2017-08-22T19:35:05.000Z,
    subObj: {
      mySubDate: 2017-08-22T19:35:05.000Z
    }
  }
 */
```

**Comment by cyrilschumacher** [comment]
Hello, thank you to have reported this mistake. I will update the project to fix this problem. ;-)

**Comment by cyrilschumacher** [comment]
After investigation, this [line](https://github.com/cyrilschumacher/json-property-filter/blob/master/src/serializer/serializeToArray.ts#L67) causes your error. I will fix this error soon.

**Comment by mkornatz** [comment]
Awesome! Thanks @cyrilschumacher. I appreciate your quick response and the fact that you open sourced your module. It's a helpful one!

**Comment by cyrilschumacher** [comment]
I have released the version: [1.3.0](https://github.com/cyrilschumacher/json-property-filter/releases/tag/1.3.0). If you find other bugs, don't hesitate to create a ticket. 😉
